src/scriv/collect.py

```diff
@@ -128,8 +128,20 @@ def collect(
     sections = order_dict(sections, [None] + config.categories)
 
     changelog = Path(config.output_file)
+    newline = None  # If no changelog exists, enable universal newlines.
     if changelog.exists():
-        changelog_text = changelog.read_text()
+        with changelog.open("r", newline="") as f:
+            changelog_text = f.read()
+        # Assume Linux newlines by default and use them internally during
+        # processing, but use Windows or Mac newlines if that's the newline
+        # style used in the existing changelog file.
+        newline = "\n"  # Linux
+        if changelog_text.count("\r\n"):  # Windows
+            newline = "\r\n"
+            changelog_text = changelog_text.replace("\r\n", "\n")
+        elif changelog_text.count("\r"):  # Mac
+            newline = "\r"
+            changelog_text = changelog_text.replace("\r", "\n")
```

Comment:
Maybe I don't understand the details, but this feels like a re-implementation of Python's universal newline mode, where it will convert newlines to \n, but keep track of what the original newlines were: https://docs.python.org/3/library/io.html#io.TextIOBase.newlines

Reply:
Also, if we have to do this ourselves like this, do we really need Mac-style \r newlines?  I don't think anyone uses them anymore.

Reply:
Doggone it, I was sure I had seen an attribute like that years ago but couldn't find the docs for it. My memory is that I checked the `dir()` output on an open file object, too, but I must have overlooked it. Yeah, way better, and thanks for pointing this out! I'll update the code.

Reply:
Updated. Note that the new code triggers an error in MyPy:

```Argument 1 to "replace" of "str" has incompatible type "Optional[str]"; expected "str"```

I guarded against `file.newlines` being None, str, or a tuple of strings. Do you have a suggestion how to resolve this?

Reply:
This is simplified now and triggers no errors. The MyPy error only triggered for Python 3.8; no other Python version in the GitHub Actions had this error so it may be a bug in MyPy.